src/lib/bootloader/stage1.rb

```diff
@@ -196,203 +218,249 @@ def available_partitions(res)
       end
       res[:extended] = extended_partition if logical_boot?
     end
+  end
 
-    def propose_x86
-      selected_location = propose_boot_location
-      log.info "propose_x86 (#{selected_location})"
-
-      # set active flag, if needed
-      if selected_location == :mbr
-        # We are installing into MBR:
-        # If there is an active partition, then we do not need to activate
-        # one (otherwise we do).
-        # Reason: if we use our own MBR code, we do not rely on the activate
-        # flag in the partition table to boot Linux. Thus, the activated
-        # partition can remain activated, which causes less problems with
-        # other installed OSes like Windows (older versions assign the C:
-        # drive letter to the activated partition).
-        used_disks = Yast::BootStorage.underlaying_devices(Yast::BootStorage.mbr_disk)
-        need_activate = used_disks.any? { |d| Yast::Storage.GetBootPartition(d).empty? }
-        self.activate = need_activate
-        self.generic_mbr = false
-      else
-        # if not installing to MBR, always activate (so the generic MBR will
-        # boot Linux)
-        self.activate = true
-        self.generic_mbr = true
-      end
+  # Represents object that can set passed stage1 to proposed values
+  class Stage1Proposal
```

Comment:
On other hand it is specialized class, that should be used only from this class. That is reason why I still keep using #proposal method in Stage1 to hide this one. And it is highly coupled to Stage1 class. It do not make sense to have it without Stage1.